frontend/pages/workspaces/[workspaceId]/folders/create-folder.tsx

```
@@ -0,0 +1,194 @@
+import React, { FC, useState } from "react";
+
+import { GraphQLClient } from "graphql-request";
+import { GetServerSideProps } from "next";
+import { useRouter } from "next/dist/client/router";
+import { Input, Form, Button } from "nhsuk-react-components";
+import { useForm } from "react-hook-form";
+import styled from "styled-components";
+
+import { Header } from "../../../../components/Header";
+import { MainHeading } from "../../../../components/MainHeading";
+import { Navigation } from "../../../../components/Navigation";
+import { PageLayout } from "../../../../components/PageLayout";
+import { Textarea } from "../../../../components/Textarea";
+import { requireAuthentication } from "../../../../lib/auth";
+import { getSdk, Folder } from "../../../../lib/generated/graphql";
+
+export const getServerSideProps: GetServerSideProps<Props> = requireAuthentication(
+  async (context) => {
+    const client = new GraphQLClient(
+      "http://workspace-service.workspace-service/graphql"
+    );
+    const sdk = getSdk(client);
+    const workspaceId = (context.params?.workspaceId as string) || "";
```

I feel like we should throw an error if the workspaceId is not set. It will error anyway because we won't be able to find a workspace. I think throwing an error explicitly makes this clearer.